File: backend/expense-tracker/src/main/java/com/college/expense_tracker/service/ExpenseService.java
package com.college.expense_tracker.service;

import com.college.expense_tracker.model.Expense;
import com.college.expense_tracker.repository.ExpenseRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ExpenseService {

    @Autowired
    private ExpenseRepository expenseRepository;

    // CREATE or UPDATE
    public Expense saveExpense(Expense expense) {
        return expenseRepository.save(expense);
    }

    // READ ALL
    public List<Expense> getAllExpenses() {
        return expenseRepository.findAll();
    }

    // READ BY ID
    public Expense getExpenseById(Long id) {
        return expenseRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Expense not found with id: " + id));
    }

    // DELETE
    public void deleteExpense(Long id) {
        expenseRepository.deleteById(id);
    }
}

File: frontend/src/services/ExpenseService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/expenses";

class ExpenseService {
  // Get all expenses
  getAllExpenses() {
    return axios.get(API_URL);
  }

  // Create a new expense
  createExpense(expense) {
    return axios.post(API_URL, expense);
  }

  // Delete an expense by ID
  deleteExpense(id) {
    return axios.delete(`${API_URL}/${id}`);
  }

  // ✅ Update an existing expense by ID
  updateExpense(id, expense) {
    return axios.put(`${API_URL}/${id}`, expense);
  }
}

// ✅ Assign instance to variable
const expenseService = new ExpenseService();

export default expenseService;
